tests/Doctrine/Tests/DBAL/ConnectionTest.php

```diff
@@ -403,13 +403,12 @@ public function testFetchColumn()
         $this->assertSame($result, $conn->fetchColumn($statement, $params, $column, $types));
     }
 
-    public function testConnectionIsClosed()
+    public function testGetPlatformVersionDoesNotFailOnBadConnection()
     {
-        $this->_conn->close();
-
-        $this->setExpectedException('Doctrine\\DBAL\\Exception\\DriverException');
-
-        $this->_conn->quoteIdentifier('Bug');
+        // the connection uses bogus connection requirements, so connecting
+        // *will* fail. We should be able to call this without connection
+        // and causing that exception
+        $this->_conn->getDatabasePlatform();
```

Comment:
> began throwing an exception if a connection couldn't be made (instead of using the default platform, like in earlier versions of DBAL).

Yeah, well, it's due to platform detection, as already discussed.

I'm also not very qualified to talk about the DBAL: it's mainly @deeky666's competence area.